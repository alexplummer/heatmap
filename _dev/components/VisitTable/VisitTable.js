
// VisitTable
// ============
// A table to track number of visitors

// Imports
import React from 'react';

class VisitTable extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            loaded: false
        }

        this.locationsList = [];
        this.winningLocation = [];
        this.leaderboardHeader = "";
    }

    componentWillUpdate(nextProps) {

        if (nextProps.locations !== this.props.locations) {

            this.setState({
                loaded: false
            });
        }
    }

    componentDidUpdate(prevProps) {

        if (prevProps.locations !== this.props.locations) {

            setTimeout(() => {
                this.setState({
                    loaded: true
                });
            }, 500);
        }
    }

    render() {

        // Render locations to table
        this.winningLocation = this.props.winner.map((thislocation) => {
            return <tr key={thislocation}><td>{thislocation[0]}</td><td>{thislocation[1]}</td></tr>
        });

        if (this.props.locations.length > 0) {
            this.leaderboardHeader = <h4>Leaderboard</h4>;

            this.locationsList = this.props.locations.map((thislocation) => {
                return <tr key={thislocation}><td>{thislocation[0]}</td><td>{thislocation[1]}</td></tr>
            });
        }

        return (
            <div className="VisitTable">
                <h3>Current winner</h3>
                <table className={this.state.loaded ? 'loaded' : ''}>
                    <tr><th>Name</th><th>Visits</th></tr>
                    {this.winningLocation}
                </table>
                {this.leaderboardHeader}
                <table>
                    {this.locationsList}
                </table>
            </div>
        );
    }
}

// Exports
export default VisitTable;
