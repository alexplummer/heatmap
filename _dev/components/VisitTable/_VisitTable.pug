
//- VisitTable
//- ============
//- A table to track number of visitors

//- Includes

mixin VisitTable()
    
    .VisitTable
