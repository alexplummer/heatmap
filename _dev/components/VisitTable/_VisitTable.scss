
// VisitTable
// ============
// A table to track number of visitors

.VisitTable {

    table {
        @extend .pure-table;
        @extend .pure-table-horizontal;
        width: 100%;

        &:first-of-type {

            tr:first-of-type{
                background: rgba(250,250,250, .2);
            }
            td {
                color: #ffe600;
            }
        }
        &:first-of-type tr:nth-child(2) td:first-child {
            @extend .icon;
            @extend .icon-star-filled;
            position: relative;

            &:before {
                position: absolute;
                left: -25px;
                top: 4px;
                font-size: $font-xl;
                color: #ffe600;
            }
        }
        &.loaded tr:nth-child(2) td:first-child:before {
            @extend .animated;
            @extend .bounce;
        }
        th, td {
            text-align: center;
        }
        th:first-of-type, td:first-of-type {
            text-align: left;
            width: 65%;
        }
        &:nth-of-type(2) {

            th, td {
                font-size: $font-xs;
            }
        }
    }
    h3, h4, th, td {
        color: #fff;
    }
    h3, h4 {
        font-weight: 300;
        margin-bottom: -$size-s;
        text-align: center;
    }
    h4 {
        text-transform: uppercase;
    }
}