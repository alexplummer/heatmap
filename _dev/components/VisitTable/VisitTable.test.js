
// MainHeader
// ============
// Heading for the document

// Imports
import React from 'react';
import renderer from 'react-test-renderer';
import VisitTable from './VisitTable';

describe('VisitTable renders correctly', () => {
    it('renders correctly', () => {
        const winner = ['location1',1];
        const locations = { 'location1': 1, 'location2': 2 };

        const rendered = renderer.create(
            <VisitTable winner={winner} locations={locations} />
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    });
});