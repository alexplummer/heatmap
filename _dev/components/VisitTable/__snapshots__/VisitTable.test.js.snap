// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisitTable renders correctly renders correctly 1`] = `
<div
  className="VisitTable"
>
  <h3>
    Current winner
  </h3>
  <table
    className=""
  >
    <tr>
      <th>
        Name
      </th>
      <th>
        Visits
      </th>
    </tr>
    <tr>
      <td>
        l
      </td>
      <td>
        o
      </td>
    </tr>
    <tr>
      <td />
      <td />
    </tr>
  </table>
  
  <table />
</div>
`;
